lib/components/FileAttachment/Attachments.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+
+import { MenuVertical, File } from "@bigbinary/neeto-icons";
+
+import Dropdown from "components/Common/Dropdown";
+import Input from "components/Common/Input";
+import MenuButton from "components/Common/MenuButton";
+
+import { DROP_DOWN_OPTIONS_KEYS, KEYS } from "./constant";
+import { getInvalidFileTypeErrorMessage, isValidFileName } from "./utils";
+
+const Attachments = ({ uploadedFiles = [], handleRename, dropDownOptions }) => {
+  const { Menu, MenuItem } = Dropdown;
+  const [fileNameError, setFileNameError] = useState("");
+  const [activeFile, setActiveFile] = useState({});
+
+  const onMenuItemClick = ({ key, handler, file }) => {
+    if (key === DROP_DOWN_OPTIONS_KEYS.RENAME) {
+      setActiveFile(file);
+    } else {
+      handler(file);
+    }
+  };
+
+  const handleKeyUp = (e, file) => {
+    if (e.key === KEYS.ENTER && handleRename) {
+      handleRename(
+        {
+          signedId: file.response.signed_id,
+          fileName: activeFile.name,
+        },
+        done => {
+          if (done) {
+            setActiveFile({});
+          }
+        }
+      );
+    }
+  };
+
+  return (
+    <div className="flex flex-wrap gap-4 my-2">
+      {uploadedFiles.map((file, index) => (
+        <div
+          className="border-2 py-1 rounded text-sm flex justify-between items-center min-w-96 width-full"
```

Comment:
1. neetoEditor is not using TailwindCSS class internally. Please replace it with SCSS classes. You can create a new file called `_file-attachments.scss` and add these styles there.
2. Please extract the mapped component to a new file (`Attachment.jsx`). Handle the error and value states within that component. Since this change will make the `Attachments.jsx` file very small, we can move the reset of the logic to the `index.jsx` file itself.
3. There will be a better key than `file${index}`. Check if the `signedId` is available. Check for similar usages.